### DISCUSSION TITLE
query error

### DISCUSSION THREAD

**Discussion by apujayed** [pocketbase/pocketbase#4023] [Q&A] [answered]
```
SELECT
	CustomerId,
	Year,
	Total,
	LAG ( Total, 1, 0 ) OVER ( 
		ORDER BY Year 
	) PreviousYearTotal 
FROM
	CustomerInvoices 
WHERE
	CustomerId = 4;
```

error - Invalid query - invalid identifier parts [LAG ( Total, 1, 0 ) OVER ( _discard_ BY Year 
	) PreviousYearTotal].

![SQLite-LEAD-function-sample-table](https://github.com/pocketbase/pocketbase/assets/95185551/035e173a-16a6-4a66-997c-056ff9261ce6)

output -
![SQLite-LAG-Function-ORDER-BY-clause-example](https://github.com/pocketbase/pocketbase/assets/95185551/c47fc863-4ca7-44bd-a5c2-f43a32ec3036)

**Comment by ganigeorgiev** [comment, marked as answer]
Try to wrap the expression in parenthesis:
```sql
(LAG(Total, 1, 0) OVER (ORDER BY Year)) PreviousYearTotal 
```
